nvflare/private/fed/app/fl_conf.py

```diff
@@ -103,6 +105,9 @@ def finalize_config(self, config_ctx: ConfigContext):
         if not secure_train:
             self.enable_byoc = True
 
+        if self.app_validator is None:
+            self.app_validator = FLAppValidator()
```

Comment:
Even if app_validator is not None, we still want to apply FLAppValidator to ensure the basic structure of the app.

I suggest:

1. Change the name of FLAppValidator to AppStructureValidator.
2. Write FLAppValidator that can take additional custom validators:

`
class FLAppValidator(AppValidator):

    def __init__(self, custom_validators: list):
         self.validators = [AppStructureValidator()]
         if custom_validators:
              self.validators.extend(custom_validators)

    def validate(self, app_folder: str) -> Tuple[str, Dict]:
          final_result = {}
          for v in self.validators:
              err, result = v.validate(app_folder)
              if err:
                  return err, result
              if result:
                  final_result.update(result)
          return "", final_result
`
3. custom_vals = [cust_val] if cust_val else []
4. self.validator = FLAppValidator(custom_vals)